producer.go

```diff
@@ -342,12 +342,21 @@ func (w *Producer) popTransaction(frameType int32, data []byte) {
 }
 
 func (w *Producer) transactionCleanup() {
-	// clean up transactions we can easily account for
-	for _, t := range w.transactions {
-		t.Error = ErrNotConnected
-		t.finish()
+l:
+	for {
+		select {
+		case data := <-w.responseChan:
```

Comment:
I think we probably have the same race in `w.errorChan`, so we should flush that channel too.

Reply:
I think you're right. The **real error reason** should pass to user, not just set  **error reason** to ```ErrNotConnected```. I will add this code.